### PR TITLE
Resolve JARs dynamically

### PR DESCRIPTION
**Pull request by Scrumplex**
This should improve the way PolyMC resolves jars such as JavaCheck or NewLaunch

### CODE REVIEW

File: launcher/Application.cpp [hunk 1]
```diff
@@ -1556,13 +1552,22 @@ shared_qobject_ptr<Meta::Index> Application::metadataIndex()
     return m_metadataIndex;
 }
 
-QString Application::getJarsPath()
+QString Application::getJarPath(QString jarFile)
 {
-    if(m_jarsPath.isEmpty())
+    QStringList potentialPaths = {
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
```
Comment (flowln) [inline launcher/Application.cpp]:
```suggestion
#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined (Q_OS_OPENBSD)
```

File: launcher/ui/pages/instance/InstanceSettingsPage.cpp [hunk 2]
```diff
@@ -50,9 +50,11 @@
 #include "Application.h"
 
 #include "java/JavaInstallList.h"
+#include "java/JavaUtils.h"
 #include "FileSystem.h"
 
 
+
```
Comment (flowln) [inline launcher/ui/pages/instance/InstanceSettingsPage.cpp]: two lines weren't enough :pensive:

File: launcher/Application.cpp [hunk 3]
```diff
@@ -1556,13 +1552,22 @@ shared_qobject_ptr<Meta::Index> Application::metadataIndex()
...
+    };
+    for(QString p : potentialPaths)
     {
-        return FS::PathCombine(QCoreApplication::applicationDirPath(), "jars");
+        QString jarPath = FS::PathCombine(p, jarFile);
+        if (QFileInfo(jarPath).isFile())
+            return jarPath;
     }
-    return FS::PathCombine(m_rootPath, m_jarsPath);
+    return QString();
```
Comment (flowln) [inline launcher/Application.cpp]:
```suggestion
    return {};
```
nitpick :)

**Review by flowln — COMMENTED** [review]
finally able to launch minecraft from the dev build without nasty hacks? :pleading_face:

**Review by timoreo22 — APPROVED** [review]

**Review by flowln — APPROVED** [review]
:+1:

**Review by d-513 — APPROVED** [review]

**Pull request merged**